src/test/java/com/wikia/webdriver/common/testnglisteners/BrowserAndTestEventListener.java

```diff
@@ -50,115 +46,130 @@ public void beforeNavigateTo(String url, WebDriver driver) {
+          // A fast lane to stop executing any test on "not a valid community" page
+          if (driver.getCurrentUrl().contains(URLsContent.NOT_A_VALID_COMMUNITY)) {
+            throw new SkipException(String.format("Wrong redirect to: %s", driver.getCurrentUrl()));
+          }
+          if (driver.getCurrentUrl().contains("data:text/html,chromewebdata ")) {
+            driver.get(url);
+            Log.warning(command, driver.getCurrentUrl());
+          } else {
+            Log.warning(command, driver.getCurrentUrl());
+          }
         }
+      } else {
+        Log.warning("Url after navigation", "Unable to check URL after navigation - alert present");
       }
-    } else {
-      Log.warning("Url after navigation", "Unable to check URL after navigation - alert present");
-    }
-
-    if (driver.getCurrentUrl().contains(Configuration.getWikiaDomain())) {
-      // HACK FOR DISABLING NOTIFICATIONS
-      try {
-        new JavascriptActions(driver).execute("$(\".sprite.close-notification\")[0].click()");
-      } catch (WebDriverException e) {
-        Log
-            .info("Hack for disabling notifications", "Failed to execute js action");
-      }
-      /**
-       * All of tests should be executed as an user who opted in (agreed) on using ads tracking.
-       * Manually user would need to click 'agree' in the tracking opt in modal.
-       */
 
-      if (TestContext.isFirstLoad()) {
-        boolean userOptedIn = true;
-        boolean userOptedOut = false;
-
-        if (method.isAnnotationPresent(Execute.class) && !method.getAnnotation(Execute.class)
-            .trackingOptIn()) {
-          userOptedIn = false;
+      if (driver.getCurrentUrl().contains(Configuration.getWikiaDomain())) {
+        // HACK FOR DISABLING NOTIFICATIONS
+        try {
+          new JavascriptActions(driver).execute("$(\".sprite.close-notification\")[0].click()");
+        } catch (WebDriverException e) {
+          Log
+              .info("Hack for disabling notifications", "Failed to execute js action");
         }
-
-        if (method.isAnnotationPresent(Execute.class) && method.getAnnotation(Execute.class)
-            .trackingOptOut()) {
-          userOptedOut = true;
+        /**
+         * All of tests should be executed as an user who opted in (agreed) on using ads tracking.
+         * Manually user would need to click 'agree' in the tracking opt in modal.
+         */
+
+        if (TestContext.isFirstLoad()) {
+          boolean userOptedIn = true;
+          boolean userOptedOut = false;
+
+          try {
+            JavascriptExecutor js = DriverProvider.getActiveDriver();
+            Object mobileWikiVersion = js.executeScript(
+                "return requirejs.entries['mobile-wiki/config/environment'].module.exports.default.APP.version");
+            Configuration.setTestValue("mobileWikiVersion", mobileWikiVersion.toString());
+          } catch (WebDriverException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)